src/server/injected/componentUtils.ts

```diff
@@ -0,0 +1,179 @@
+    if (attr.op === '*=')
+      return attr.caseSensetive ? obj.includes(attr.value) : obj.toUpperCase().includes(attr.value);
+  }
+  if (attr.op === '=')
+    return obj === attr.value;
+  return false;
+}
+
+export function parseComponentSelector(selector: string): ParsedComponentSelector {
+  let wp = 0;
+  let EOL = selector.length === 0;
+  let next = selector.length ? selector[0] : '';
+  const eat1 = () => {
+    const result = next;
+    ++wp;
+    next = wp < selector.length ? selector[wp] : '';
+    EOL = wp >= selector.length;
+    return result;
+  };
+  const syntaxError = (name: string) => {
+    if (EOL)
+      throw new Error(`Unexpected end of selector \`${selector}\` while parsing "${name}"`);
+    throw new Error(`Error while parsing selector \`${selector}\` - unexpected symbol "${next}" at position ${wp} while parsing ${name}`);
+  };
+
+  function skipSpaces() {
+    while (!EOL && /\s/.test(next)) eat1();
+  }
+
+  function readIdentifier() {
+    let result = '';
+    while (!EOL && /[-$0-9A-Z_]/i.test(next))
+      result += eat1();
+    return result;
+  }
+
+  function readQuotedString(quote: string) {
+    let result = eat1();
+    if (result !== quote)
+      syntaxError('quoted string');
+    while (!EOL && next !== quote)
+      result += eat1();
+    if (next !== quote)
+      syntaxError('quoted string');
+    result += eat1();
+    return result;
+  }
+
+  function readAttributeToken() {
+    skipSpaces();
+    let token = '';
+    if (next === `'` || next === `"`)
+      token = readQuotedString(next).slice(1, -1);
+    else
+      token = readIdentifier();
+    skipSpaces();
+    if (!token)
+      syntaxError('property path');
+    return token;
+  }
+
+  function readOperator(): '='|'*=' {
+    let op = '';
+    if (!EOL)
+      op += eat1();
+    if (!EOL && (op === '*'))
+      op += eat1();
+    if (op !== '=' && op !== '*=')
+      syntaxError('operator');
+    return (op as any);
+  }
+
+  function readAttribute(): ParsedComponentAttribute {
+    // skip leading [
+    eat1();
+
+    // read attribute name:
+    // foo.bar
+    // 'foo'  . "ba zz"
+    const jsonPath = [];
+    jsonPath.push(readAttributeToken());
+    while (next === '.') {
+      eat1();
+      jsonPath.push(readAttributeToken());
+    }
+
+    skipSpaces();
+    const operator = readOperator();
+
+    skipSpaces();
+    let valueText = '';
+    if (next === `'` || next === `"`) {
+      valueText = readQuotedString(next);
+    } else {
+      while (!EOL && !/\s/.test(next) && next !== ']')
+        valueText += eat1();
+    }
+    skipSpaces();
+    let caseSensetive = true;
+    if (next === 'i' || next === 'I') {
+      caseSensetive = false;
+      eat1();
+    } else if (next === 's' || next === 'S') {
+      caseSensetive = true;
+      eat1();
+    }
+    skipSpaces();
+    if (next !== ']')
+      syntaxError('attribute value');
+    let value;
+    try {
+      value = eval(valueText);
```

Comment:
Why eval? Do we support random values? I haven't seen the tests for this. Looks scary.

Reply:
this helps parsing numbers, float numbers, undeffineds, nulls, Infinity, e.t.c. This is a little scary, but I'd keep it as-is for now for simplicity unless there's a report